compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/dfa/cfg/ControlFlowGraphBuilder.kt

```diff
@@ -804,7 +807,15 @@ class ControlFlowGraphBuilder {
         levelCounter--
         val node = createTryMainBlockExitNode(tryExpression)
         popAndAddEdge(node)
-        addEdge(node, tryExitNodes.top())
+        // NB: If this block can escape, not needed to connect to the finally block.
+        val tryMainBlockEscaped = blockEscaped(node) { n -> n is TryMainBlockEnterNode && n.level == levelCounter }
```

Comment:
I still didn't get why we need such condition and what problems it solves

Reply:
So, in your diagram in my previous PR, consider the path from `catch 2` through `finally` to the function exit. Without this condition, such path will be flown to the `finally` and needs to flow out to the function exit. This needs either handling multiple edges with the same label or labeling edges on that path with another unique label. With this condition, we can avoid adding a spurious path in the beginning, which also makes the path-sensitive version of CFG traversal as well as the addition of path-aware property init info a lot simpler, because we can focus on only normal v.s. uncaught-exception paths.

Reply:
continued at the other comment.